### PR TITLE
[Workers] fix vulnerability in signature example

### PR DESCRIPTION
**Pull request by tniessen**
Without proper separation/padding, the existing example potentially allows bypassing signature/expiry checks.

For example, a signature for the pathname `"/get/123456"` with expiry `1643291710779` would also be valid for the pathname `"/get/12345"` with expiry `61643291710779`, thus extending the signature's validity by two millennia, while the pathname might still be valid. Similarly, if no access to `"/get/12345"` was intended, this allows privilege escalation.

### CODE REVIEW

**Review by kristianfreeman — APPROVED** [review]

File: products/workers/src/content/examples/signing-requests.md [hunk 1]
```diff
@@ -54,9 +54,11 @@ async function verifyAndFetch(request) {
 
   // Extract the query parameters we need and run the HMAC algorithm on the
   // parts of the request we are authenticating: the path and the expiration
-  // timestamp.
+  // timestamp. It is crucial to pad the input data, e.g., by adding a symbol
```
Comment (deadlypants1973) [inline products/workers/src/content/examples/signing-requests.md]:
```suggestion
  // timestamp. It is crucial to pad the input data, for example, by adding a symbol
```

File: products/workers/src/content/examples/signing-requests.md [hunk 2]
```diff
@@ -54,9 +54,11 @@ async function verifyAndFetch(request) {
...
+  // in-between the two fields that can never occur on the right side. Inthis
```
Comment (deadlypants1973) [inline products/workers/src/content/examples/signing-requests.md]:
```suggestion
  // in-between the two fields that can never occur on the right side. In this
```

File: products/workers/src/content/examples/signing-requests.md [hunk 3]
```diff
@@ -54,9 +54,11 @@ async function verifyAndFetch(request) {
...
+  // case, we use the @ symbol to separate the fields.
```
Comment (deadlypants1973) [inline products/workers/src/content/examples/signing-requests.md]:
```suggestion
  // case, use the @ symbol to separate the fields.
```

File: products/workers/src/content/examples/signing-requests.md [hunk 4]
```diff
@@ -130,7 +132,14 @@ async function generateSignedUrl(url) {
   // parameter.
   const expirationMs = 60000
   const expiry = Date.now() + expirationMs
-  const dataToAuthenticate = url.pathname + expiry
+  // The signature will be computed for the pathname and the expiry timestamp.
+  // The two fields must be separated or padded to ensure that an attacker
+  // will not be able to use the same signature for other pathname/expiry pairs.
+  // The @ symbol is guaranteed not to appear in expiry, which is a (decimal)
+  // number, so we can safely use it as a separator here. When combining more
```
Comment (deadlypants1973) [inline products/workers/src/content/examples/signing-requests.md]:
```suggestion
  // number, so you can safely use it as a separator here. When combining more
```

**Review by deadlypants1973 — COMMENTED** [review]
PCX style guide edits :)

**Review by deadlypants1973 — APPROVED** [review]

**Pull request merged**